src/components/DataActionButton/index.tsx

```diff
@@ -64,6 +86,182 @@ function DataActionButton(props: Props) {
+  }
+
+  const idx = encodeURIComponent(idxFilePresignUrl);
+  const enf = encodeURIComponent(filePresignUrl);
+  const name = pathOrKey.split('/').pop() ?? pathOrKey;
+  return `http://localhost:60151/load?index=${idx}&file=${enf}&name=${name}`;
+};
+
+const constructS3LocalIgvUrl = async (props: {
+  id: number;
+  bucketOrVolume: string;
+  pathOrKey: string;
+}) => {
+  const { id, bucketOrVolume, pathOrKey } = props;
+
+  const objStatus = await getS3Status(id);
+  if (objStatus != S3StatusData.AVAILABLE) {
+    throw new Error(`Invalid object storage class! (${S3StatusData[objStatus]})`);
+  }
+
+  const name = pathOrKey.split('/').pop() ?? pathOrKey;
+  const file = `s3://${bucketOrVolume + '/' + pathOrKey}`;
+
+  return `http://localhost:60151/load?file=${encodeURIComponent(file)}&name=${name}`;
+};
+
+type OpenIGVDesktopType = DataActionButtonProps & { handleIsOpen: (val: boolean) => void };
+function OpenIGVDesktop(props: OpenIGVDesktopType) {
+  const toast = useToastContext();
+
+  const { id, bucketOrVolume, pathOrKey, type, handleIsOpen } = props;
+
+  const gdsLocalIgvUrl = useQuery(
+    ['gds-local-igv', bucketOrVolume, pathOrKey],
+    async () =>
+      await constructGDSLocalIgvUrl({ bucketOrVolume: bucketOrVolume, pathOrKey: pathOrKey }),
+    { enabled: type == 'gds' }
+  );
+
+  const s3LocalIgvUrl = useQuery(
+    ['s3-local-igv', bucketOrVolume, pathOrKey],
+    async () =>
+      await constructS3LocalIgvUrl({
+        id: id,
+        bucketOrVolume: bucketOrVolume,
+        pathOrKey: pathOrKey,
+      }),
+    { enabled: type == 's3' }
+  );
+
+  if (gdsLocalIgvUrl.isLoading || s3LocalIgvUrl.isLoading) {
+    return (
+      <Dialog
+        header='Opening in IGV Desktop'
+        visible={true}
+        style={{ width: '50vw' }}
+        draggable={false}
+        resizable={false}
+        onHide={() => handleIsOpen(false)}>
+        <CircularLoaderWithText />
+      </Dialog>
+    );
+  }
+
+  if (s3LocalIgvUrl.isError && s3LocalIgvUrl.error) {
+    toast?.show({
+      severity: 'error',
+      summary: 'Error on locating object URL.',
+      detail: `${s3LocalIgvUrl.error}`,
+      sticky: true,
+    });
+  }
+  if (gdsLocalIgvUrl.isError && gdsLocalIgvUrl.error) {
+    toast?.show({
+      severity: 'error',
+      summary: 'Error on locating object URL.',
+      detail: `${gdsLocalIgvUrl.error}`,
+      sticky: true,
+    });
+  }
+
+  const xhr = new XMLHttpRequest();
+
+  let localIgvUrl = '';
+
+  localIgvUrl = type == 'gds' && gdsLocalIgvUrl.data ? gdsLocalIgvUrl.data : '';
+  localIgvUrl = type == 's3' && s3LocalIgvUrl.data ? s3LocalIgvUrl.data : '';
+
+  xhr.open('GET', localIgvUrl, true);
+  xhr.onreadystatechange = () => {
+    if (xhr.readyState === 4 && xhr.status === 0) {
+      const message =
+        'Cannot open automatically in IGV. Please make sure you have opened IGV app and try again. ' +
+        'Otherwise please click "Copy" button and open the URL in browser new tab.';
+
+      <Dialog
```

Comment:
This might not work, yet. Need to confirm. Dialog will be probably out of scope within XHR callback block. 

But. Happy to go pass this for now and, push it through for first cut. We can revisit specifically with another follow up fix PR.

Reply:
We can deal with this tickie later -- https://github.com/umccr/data-portal-client/issues/172

Reply:
Hmm,  I am changing this now to put this inside a Toast instead. It seems to be working, but just need some styling to the component.

Reply:
Yep. We can just simply close it 172 if go with it now; so no worries.

Reply:
I just pushed to commit 2369187. This should solve it.